src/BarrelHCalCalorimeter_geo.cpp

```diff
@@ -0,0 +1,481 @@
+//==========================================================================
+//  AIDA Detector description implementation
+//--------------------------------------------------------------------------
+// Copyright (C) Organisation europeenne pour la Recherche nucleaire (CERN)
+// All rights reserved.
+//
+// For the licensing terms see $DD4hepINSTALL/LICENSE.
+// For the list of contributors see $DD4hepINSTALL/doc/CREDITS.
+//
+// Author     : M.Frank
+//
+//==========================================================================
+//
+// Specialized generic detector constructor
+//
+//==========================================================================
+#include "DD4hep/DetFactoryHelper.h"
+#include "XML/Layering.h"
+#include "DD4hep/Printout.h"
+
+#include "TVector3.h"
+
+using namespace std;
+using namespace dd4hep;
+using namespace dd4hep::detail;
+
+static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector sens)
+{
+
+  //printout(WARNING, "BarrelHCalCalorimeter", "called create_detector ");
+
+  xml_det_t     x_det             = e;
+  int           det_id            = x_det.id();
+  string        det_name          = x_det.nameStr();
+  Material      air               = description.air();
+
+  // get the solids section for this detector
+  xml_comp_t    x_solids         = x_det.child("solids"); 
+
+  DetElement       sdet(det_name, det_id);
+  Volume           motherVol = description.pickMotherVolume(sdet);
+
+  // Create envelope to hold HCAL barrel 
+
+  double rmin1 = x_det.rmin1(); 
+  double rmin2 = x_det.rmin1(); 
+  double rmax = x_det.rmax(); 
+  double length1 = x_det.z1(); 
+  double length2 = x_det.z2();
+  
+  std::vector<double> rmins; 
+  rmins[0] = rmin1; 
+  rmins[1] = rmin2; 
```

Comment:
Suggestion to use an initializer list (since c++11):
```suggestion
  std::vector<double> rmins = {rmin1, rmin2};
```
Same with `rmaxs` and `zs` below.

Reply:
Done. Previous version passed overlap checks for epic.xml.  Looking forward to a lean bill of health from the next set of checks....